### ISSUE TITLE
colon_rule needs two passes to detect violations in generics

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2664]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

If I add this snippet to `triggeringExamples`, `colon_rule` tests fail because it only detects one of the violations.

```swift
class ↓Foo<↓T : Equatable> : Bar {}
```

The implication of that is that two `swiftlint autocorrect` passes will be needed to fix both violations.

This was introduced on #2633
